builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -118,3 +157,73 @@ func resourceAwsS3BucketDelete(d *schema.ResourceData, meta interface{}) error {
 	}
 	return nil
 }
+
+func updateWebsite(s3conn *s3.S3, d *schema.ResourceData) error {
+	website := d.Get("website").(bool)
+	bucket := d.Get("bucket").(string)
+	indexDocument := d.Get("index_document").(string)
+	errorDocument := d.Get("error_document").(string)
```

Comment:
There is a method to check for changes, `d.HasChange`, which can help you here. As it stands now, every update will do `PutBucketWebsiteInput` if `website=true`. If `website=false`, you still call `s3conn.DeleteBucketWebsite()` every time. Ex, you create a s3 bucket but do not specify `website=true`, that API is still called.